### PR TITLE
[enhancement](mysql) enable two-way ssl authentication

### PR DESCRIPTION
**Pull request by Xiaoccer**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

According to the [mysql-ssl](https://dev.mysql.com/doc/refman/8.0/en/creating-ssl-files-using-openssl.html), enable two-way SSL authentication.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [x] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: regression-test/ssl_default_certificate/server-req.pem
Comment (gavinchou) [inline regression-test/ssl_default_certificate/server-req.pem:1]: Keep files that are needed by the client only.

File: conf/mysql_ssl_default_certificate/ca-key.pem
Comment (gavinchou) [inline conf/mysql_ssl_default_certificate/ca-key.pem:1]: Keep files that are needed by the server only.

**Comment by gavinchou** [comment]
LGTM

**Comment by Xiaoccer** [comment]
run buildall

**Comment by morningman** [comment]
you need to add following files to `.licenserc.yaml` to pass the license check

```
conf/mysql_ssl_default_certificate/client_certificate/ca.pem
conf/mysql_ssl_default_certificate/client_certificate/client-cert.pem
conf/mysql_ssl_default_certificate/client_certificate/client-key.pem
regression-test/ssl_default_certificate/ca.pem
regression-test/ssl_default_certificate/client-cert.pem
regression-test/ssl_default_certificate/client-key.pem 
```

**Comment by Xiaoccer** [comment]
run buildall

**Comment by Xiaoccer** [comment]
run buildall

**Comment by gavinchou** [comment]
LGTM

**Review by Yukang-Lian — DISMISSED** [review]
LGTM

**Review by gavinchou — DISMISSED** [review]

**Comment by Xiaoccer** [comment]
run buildall

File: conf/mysql_ssl_default_certificate/client_certificate/client-cert.pem [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+M/Hd0PPFubpEoqg/8qjIz/TbQIXbJ2FYkKFv8Z3RYvy0GP5ZVpTm4zcYB6RAzr6z
++qbA3Li/0UjUVSdhzsoDWn5lOfX6Dp7yAuiocSMpMk65A/pwwRPSJ9u/gPP2LVsJ
+L5uBogk29Hj5QBwRGePz0hJnDR3C4Lb786zHWk0QmHvVxQJq+DIbY8vlMhv6DwID
+AQABMA0GCSqGSIb3DQEBCwUAA4IBAQCgi2pRKqWiZv6Xlpn4Viv/N+G9J+0/IUnd
+YWvhmF4yzBb4R4FjyxiKG9d79o6JhhJ1ts5fmNk/idS0sBoj8FOkj53KAbw6pHBQ
+bO3f+UYWLvx8I8F5iycAseA5GTid2cOU8s/gY34rhvey2PGzR+hxfDDGbpRxXFKw
+X4zOKCYK8qAR9dDc8MOJyAs30NXn6vxiQSNijJe7+0J91NbAOHw/NeaIS673exqs
+K7nPiAe7tPwZOY5LsZxzrosTIsUryheM8S+S0Sqess+zkKMV1xbCbyk2eMbhdfyL
+5xLGv7HxnIEoJyRKQ4q0wk9GteLdvlSAKJ1cTe/n8NOf36cXZj/s
+-----END CERTIFICATE-----
```
Comment (CalvinKirs) [inline conf/mysql_ssl_default_certificate/client_certificate/client-cert.pem:21]: There is no reason for us to provide this.

Comment (Xiaoccer) [reply conf/mysql_ssl_default_certificate/client_certificate/client-cert.pem:21]: There is a README written that these certificates cannot be used in the production environment, only for testing

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**